deep_quoridor/src/agents/greedy.py

```diff
@@ -0,0 +1,157 @@
+from queue import Queue
+
+import numpy as np
+
+from agents.core import Agent
+
+type Position = tuple[int, int]
+
+
+class GreedyAgent(Agent):
+    def __init__(self, **kwargs):
+        super().__init__()
+
+    def start_game(self, game, player_id):
+        self.player_id = player_id
+
+    def _valid_pawn_movements(self, game, action_mask: np.ndarray) -> list[Position]:
+        """Returns the coordinates of the possible movements based on the action mask.
+        This is useful for the first move, since there could be complicated jumps.  After the
+        first move, you can't easily use this because you'd need to generate a new action_mask
+        """
+        movement_mask = action_mask[: game.board_size**2]
+        valid_moves = [i for i, x in enumerate(movement_mask) if x == 1]
```

Comment:
This is fine, but if you wanted to make it a bit faster and more numpyonic you could use `np.argwhere(movement_mask==1)`. And that's the same as `np.argwhere(movement_mask)` since 0 evaluates to false (but `x==1` is clearer so i don't have a preference.

Reply:
that's a good one, I'm replacing it